src/main/java/org/folio/inventory/resources/AbstractInstances.java

```diff
@@ -109,61 +109,161 @@ protected void updateInstanceRelationships(Instance instance,
+
+    allResultsOf(allFutures).thenAccept(consumer);
+  }
+
+  protected void updatePrecedingSucceedingTitles(Instance instance, RoutingContext routingContext,
+    WebContext context, Consumer respond) {
+    CollectionResourceClient precedingSucceedingTitlesClient = createPrecedingSucceedingTitlesClient(routingContext, context);
+    List<String> instanceId = Arrays.asList(instance.getId());
+    String query = createQueryForPrecedingSucceedingInstances(instanceId);
+
+    if (precedingSucceedingTitlesClient != null) {
+      precedingSucceedingTitlesClient.getMany(query, (Response result) -> {
+        List<CompletableFuture<Response>> allFutures = new ArrayList<>();
+        if (result.getStatusCode() == 200) {
+          JsonObject json = result.getJson();
+          List<JsonObject> relationsList = JsonArrayHelper.toList(json.getJsonArray("precedingSucceedingTitles"));
+          Map<String, PrecedingSucceedingTitle> existingPrecedingSucceedingTitles =
+            getExistedPrecedingSucceedingTitles(relationsList);
+          Map<String, PrecedingSucceedingTitle> updatingPrecedingSucceedingTitles =
+            getUpdatingPrecedingSucceedingTitles(instance);
+
+          update(precedingSucceedingTitlesClient,
+            allFutures, existingPrecedingSucceedingTitles, updatingPrecedingSucceedingTitles);
+
+          allOf(allFutures).thenAccept(respond);
+        }
+      });
+    }
+  }
+
+  private Map<String, PrecedingSucceedingTitle> getExistedPrecedingSucceedingTitles(
+    List<JsonObject> relationsList) {
 
+    Map<String, PrecedingSucceedingTitle> existingPrecedingSucceedingTitles = new HashMap();
+    relationsList.stream().map(PrecedingSucceedingTitle::new).forEachOrdered(relObj ->
+      existingPrecedingSucceedingTitles.put(relObj.id, relObj));
+
+    return existingPrecedingSucceedingTitles;
+  }
+
+  private <T> void update(CollectionResourceClient resourceClient,
+    List<CompletableFuture<Response>> allFutures,
+    Map<String, T> existingObjects, Map<String, T> updatingObjects) {
+
+    createOrEdit(resourceClient, allFutures, existingObjects, updatingObjects);
+    delete(resourceClient, allFutures, existingObjects, updatingObjects);
+  }
+
+  private <T> void delete(CollectionResourceClient resourceClient,
+   List<CompletableFuture<Response>> allFutures, Map<String, T> existingObjects,
+   Map<String, T> updatingObjects) {
+
+    existingObjects.keySet().stream()
+      .filter(key -> !updatingObjects.containsKey(key)).forEach(existingKey -> {
+      CompletableFuture<Response> newFuture = new CompletableFuture<>();
+      allFutures.add(newFuture);
+      resourceClient.delete(existingKey, newFuture::complete);
+    });
+  }
+
+  private <T> void createOrEdit(CollectionResourceClient resourceClient,
+                                List<CompletableFuture<Response>> allFutures, Map<String, T> existingObjects,
+                                Map<String, T> updatingObjects) {
```

Comment:
Please can these lines be formatted with standard indentation rather than horizontal alignment